### PR TITLE
sql/migrate: merge Dir and Scanner to one interface

### PR DESCRIPTION
**Pull request by a8m**
The next steps are supporting `Baseline` file(s), `Tags`, and `flyway` directory.

### CODE REVIEW

File: cmd/atlas/internal/cmdapi/migrate_test.go [hunk 1]
```diff
@@ -582,3 +588,12 @@ func countFiles(t *testing.T, p string) int {
 	require.NoError(t, err)
 	return len(files)
 }
+
+func sed(t *testing.T, r, p string) {
+	args := []string{"-i"}
+	if runtime.GOOS == "darwin" {
+		args = append(args, ".bk")
+	}
```
Comment (a8m) [inline cmd/atlas/internal/cmdapi/migrate_test.go:596]: It fails locally on darwin.

Comment (masseelch) [reply cmd/atlas/internal/cmdapi/migrate_test.go:595]: Darwin has no GNU sed. You need to install it.

File: sql/migrate/migrate.go [hunk 2]
```diff
@@ -140,11 +140,14 @@ func Conn(drv Driver, opts *schema.InspectRealmOption) StateReader {
 }
 
 type (
-	// Dir describes the methods needed for a Planner to manage migration files.
+	// Dir wraps the functionality described a migration directory.
```
Comment (rotemtam) [inline sql/migrate/migrate.go:143]: That describes ?

**Review by rotemtam — APPROVED** [review]
Minor godoc fix

File: sql/migrate/migrate.go (hunk 2 above)
Comment (a8m) [inline sql/migrate/migrate.go:143]:
```suggestion
	// Dir wraps the functionality used to interact with a migration directory.
```

**Pull request merged**